universal-application-tool-0.0.1/app/services/program/ProgramServiceError.java

```diff
@@ -0,0 +1,15 @@
+package services.program;
+
+import com.google.auto.value.AutoValue;
+
+/** Represents a failure for a program service operation. */
+@AutoValue
+public abstract class ProgramServiceError {
+
+  public static ProgramServiceError of(String message) {
+    return new AutoValue_ProgramServiceError(message);
+  }
+
+  /** The failure message. */
+  public abstract String message();
+}
```

Comment:
Issue (non-blocking): this seems somewhat overly specific to program service given that it is just wrapping a string. Consider making this `services.CiviFormError`? It'd be nice to get re-use out of the code that consumes it when we can.

Reply:
Done.